internal/resource/nvml-device.go

```diff
@@ -99,3 +100,35 @@ func (d nvmlDevice) GetPCIClass() (uint32, error) {
 	}
 	return nvDevice.Class, nil
 }
+
+func (d nvmlDevice) IsImexCapable() (bool, error) {
+	return true, nil
+}
+
+func (d nvmlDevice) GetClusterUUID() (string, error) {
+	gfInfo, ret := d.GetGpuFabricInfo()
+	if ret != nvml.SUCCESS {
+		return "", ret
+	}
+
+	// Convert the array to a byte slice
+	byteSlice := gfInfo.ClusterUuid[:]
+	clusterUUID := fmt.Sprintf("%08x-%04x-%04x-%04x-%012x",
```

Comment:
Why not just return the slice as a string?

Reply:
suggestion taken

Reply:
Does this mean that you updated this?

Reply:
Also, it seems as if we could also just use:
```suggestion
	clusterUUID := uuid.FromBytes(gfInfo.ClusterUuid)
```
here?